azkaban-common/src/test/java/azkaban/executor/ExecutionFlowDaoTest.java

```diff
@@ -586,11 +565,191 @@ public void testSelectAndUpdateExecutionWithSamePriority() throws Exception {
+
+    final Map<Integer, Pair<ExecutionReference, ExecutableFlow>> activeFlows2 =
+        this.fetchActiveFlowDao.fetchActiveFlows();
+    assertThat(activeFlows2.containsKey(flow1.getExecutionId())).isTrue();
+    assertThat(activeFlows2.get(flow1.getExecutionId()).getSecond().getStatus())
+        .isEqualTo(Status.KILLING);
+  }
+
+  @Test
+  public void testFlowStatusWithFetchActiveFlowByExecId() throws Exception {
+    final ExecutableFlow flow1 = createExecutionAndAssign(Status.RUNNING,
+        this.executorDao.addExecutor("test", 1));
+    Pair<ExecutionReference, ExecutableFlow> activeFlow1 =
+        this.fetchActiveFlowDao.fetchActiveFlowByExecId(flow1.getExecutionId());
+    assertThat(activeFlow1).isNotNull();
+    assertTwoFlowSame(activeFlow1.getSecond(), flow1);
+
+    makeFlowStatusInconsistent(flow1.getExecutionId(), Status.KILLING);
+
+    Pair<ExecutionReference, ExecutableFlow> activeFlow2 =
+        this.fetchActiveFlowDao.fetchActiveFlowByExecId(flow1.getExecutionId());
+    assertThat(activeFlow2).isNotNull();
+    assertThat(activeFlow2.getSecond().getStatus()).isEqualTo(Status.KILLING);
+  }
+
+  @Test
+  public void testFlowStatusWithFetchQueuedFlows() throws Exception {
+    final ExecutableFlow flow1 = submitNewFlow("exectest1", "exec1", System.currentTimeMillis(),
+        ExecutionOptions.DEFAULT_FLOW_PRIORITY);
+    flow1.setStatus(Status.RUNNING);
+    this.executionFlowDao.updateExecutableFlow(flow1);
+
+    final List<Pair<ExecutionReference, ExecutableFlow>> fetchedQueuedFlows1 =
+        this.executionFlowDao.fetchQueuedFlows();
+    assertThat(fetchedQueuedFlows1).isEmpty();
+
+    makeFlowStatusInconsistent(flow1.getExecutionId(), Status.PREPARING);
+
+    final List<Pair<ExecutionReference, ExecutableFlow>> fetchedQueuedFlows2 =
+        this.executionFlowDao.fetchQueuedFlows();
+    assertThat(fetchedQueuedFlows2).isNotEmpty();
+    assertThat(fetchedQueuedFlows2.get(0).getSecond().getStatus()).isEqualTo(Status.PREPARING);
+  }
+
+  @Test
+  public void testFlowStatusWithFetchRecentlyFinishedFlows() throws Exception {
+    final ExecutableFlow flow1 = createExecution(System.currentTimeMillis(), Status.SUCCEEDED);
+    flow1.setEndTime(System.currentTimeMillis());
+    this.executionFlowDao.updateExecutableFlow(flow1);
+
+    final List<ExecutableFlow> finishedFlows1 = this.executionFlowDao.fetchRecentlyFinishedFlows(
+        RECENTLY_FINISHED_LIFETIME);
+    assertThat(finishedFlows1).isNotEmpty();
+    assertTwoFlowSame(flow1, finishedFlows1.get(0));
+
+    makeFlowStatusInconsistent(flow1.getExecutionId(), Status.FAILED);
+
+    final List<ExecutableFlow> finishedFlows2 = this.executionFlowDao.fetchRecentlyFinishedFlows(
+        RECENTLY_FINISHED_LIFETIME);
+    assertThat(finishedFlows2).isNotEmpty();
+    assertThat(finishedFlows2.get(0).getStatus()).isEqualTo(Status.FAILED);
+  }
+
+  private void makeFlowStatusInconsistent(int executionId, Status status) {
```

Comment:
This method caught my attention because it says "make the status inconsistent".  While it's true that we got the inconsistency because of status column being updated independent of the object we store, we may consider renaming this or add a comment? :)

Reply:
thanks, I added a comment.